### ISSUE TITLE
IMAP Error on IIS

### ISSUE THREAD

**Issue by ralekeo** [freescout-helpdesk/freescout#3293]
I have deployed successfully on IIS, however I am getting the following with Office365.

imap_open(): Couldn't open stream {outlook.office365.com:993/imap/ssl}. LOGIN failed.; LOGIN failed.; LOGIN failed.; Too many login failures; File: C:\inetpub\wwwroot\Freescout\overrides\webklex\laravel-imap\src\IMAP\Client.php (230)

The same Office365 configuration works on our test Linux server on azure, but fails on premise (IIS) with the above error. 

Clicking on the Connect button (on IIS implementation) navigates successfully to office365 to sign in and returns back to the same page without any change and no error is logged
![image](https://github.com/freescout-helpdesk/freescout/assets/26841943/187d8612-ee2b-46de-bb73-883ccad1749a)

However, on Azure Linux, it changes to Disconnect and works as expected

So I am pretty sure it is not a credential issue. Any help or direction would be appreciated

**Comment by freescout-helpdesk** [comment]
Unfortunately we don't have ideas what may cause this.

**Comment by ralekeo** [comment]
Thank you for the feedback and sorry for the late response

After digging around in the code, I was able to narrow down the problem. 

On checking the mailbox table in MySQL I discovered the Meta Column which was significantly different between the two installations (Below is the entry for the Linux installation. The windows installation was empty)

![image](https://github.com/freescout-helpdesk/freescout/assets/26841943/4fd58f34-6a70-4c6f-bc86-ebb494ef3c28)


When I enabled debug mode in Webklex files, I realized that this a_token was used to sign in into the Office365. Then, using 

 openssl s_client -crlf -connect outlook.office365.com:993 
 A1 LOGIN client_id a_token
 A2 LIST
    
I was then able to successfully log into IMAP and list the folders in the Windows Server. This eliminated Proxy, Office365, Certificate, DNS, Network, SSL and Firewall. 

I copied the meta entries from the Linux to the Windows database and it was able to connect successfully but was still unable to fetch email. 

On review of the I discovered that this [code](https://github.com/freescout-helpdesk/freescout/blob/e75dcaf282e6a94fa0221fa4ce6dc21ee10de3b4/app/Http/Controllers/MailboxesController.php#L861) was returning integers on the Windows system as that is what is used to fetch the auth code in 888 which is saved to the column in  895, I figured that is where the problem is coming from. 

Given the windows code was copied straight from the Linux environment, It was only proper to assume that the problem is related to the windows php installation. 

All the extensions returns OK and the version of PHP are the same. 

I would post further update, when I am able to resolve this or most likely work around it, given that the business is already pissed at the delay

**Comment by freescout-helpdesk** [comment]
We'll close this. If you'll find a solution please share with us.

**Comment by ralekeo** [comment]
Thank you very much,  I found the solution to it for any other person who needs it

The problem was coming from [Curl Certificate Verification](https://github.com/freescout-helpdesk/freescout/blob/9be2c1df5a14747a811156b0bc08730107f1c24f/app/Misc/Mail.php#L818)

It worked after I set this to false. The curl call comes back empty because it could not verify the SSL Certificate. I presume that this was because I was on premise.
